### DISCUSSION TITLE
Wiring Diagram

### DISCUSSION THREAD

**Discussion by npawelek** [tagyoureit/nodejs-poolController#929] [RS485] [answered]
I purchased one of the recommended [Industrial RS 485](https://www.amazon.com/gp/product/B081MB6PN2/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1) adapters and after plugging it into my equipment I'm not seeing anything when running `od`. I've ensured I have the proper serial device (`/dev/ttyUSB0`), which was confirmed via `dmesg`. I have an EasyTouch panel with about 15 feet of Cat6A running from the controller serial port (wired only to YEL and GRN), back to the serial adapter plugged into a RPi4. I noticed that the 485 adapter also has a ground, so I'm wondering if I need to wire that, which wasn't explicitly called out in the wiki. I was thinking perhaps it could be the cable length or the lack of ground connection for why I don't see anything. In the raspi-config, I also verified that both serial settings under interface are set to `no` per the wiki.

Any ideas what I may be doing wrong?

**Comment by rstrouse** [comment]
Both serial settings?  Only the serial console should be turned off.

**Reply by npawelek** [reply]
Thanks for the response. I swapped the A+ and B- wires and started seeing data that appears to be in the correct format, so that is good.  My question around wiring the GND still stands. Is it needed?

On the `raspi-config` serial options, there's a [section in the adapter portion of the wiki](https://github.com/tagyoureit/nodejs-poolController/wiki/RS-485-Adapter-Details#raspi) that states to set both to "no". Is this incorrect? Looks like it still works with it set this way.

![serial1](https://github.com/tagyoureit/nodejs-poolController/assets/4248420/59c1feb5-c853-4fc4-8411-8ed5df57ed7a)
![serial2](https://github.com/tagyoureit/nodejs-poolController/assets/4248420/241e609d-a598-4cf7-adaa-0cef3269a435)

**Reply by tagyoureit** [reply, marked as answer]
That page hasn't been updated in a long time, but was incorrect as you now see.  I changed it.

And no, you don't need to ground the RS-485 adapter.

**Reply by npawelek** [reply]
Thank you
